### PR TITLE
Allow the Service to communicate default settings for oAuth and Limits

### PR DESCRIPTION
**Pull request by borkweb**
See: FIndings doc

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Must use &quot;self::&quot; for local static member reference | [Aggregator.php:139](https://github.com/moderntribe/the-events-calendar/blob/cc65c37a73ebdcb3f6834a7190306897c023d1d5/src/Tribe/Aggregator.php#L139)
:x: | Must use &quot;self::&quot; for local static member reference | [Aggregator.php:327](https://github.com/moderntribe/the-events-calendar/blob/cc65c37a73ebdcb3f6834a7190306897c023d1d5/src/Tribe/Aggregator.php#L327)
:x: | Blank line found after control structure | [tribe-options-addons-api.php:99](https://github.com/moderntribe/the-events-calendar/blob/cc65c37a73ebdcb3f6834a7190306897c023d1d5/src/admin-views/tribe-options-addons-api.php#L99)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Aggregator.php [hunk 1]
```diff
@@ -128,13 +128,20 @@ private function __construct() {
 		// Remove aggregator records from ET
 		add_filter( 'tribe_tickets_settings_post_types', array( $this, 'filter_remove_record_post_type' ) );
 
-		// Notice users about expiring Facebook Token
+		// Notice users about expiring Facebook Token if oauth is enabled
```
Comment (barryhughes) [inline src/Tribe/Aggregator.php]: Supremely nitpicky but should this start with _Notify_ rather than _Notice?_

File: src/Tribe/Aggregator.php [hunk 2]
```diff
@@ -128,13 +128,20 @@ private function __construct() {
...
+		add_action( 'plugins_loaded', array( $this, 'setup_notices' ), 11 );
+	}
 
-		/**
-		 * @todo  fb-reviewed-reactivation
-		 */
-		// tribe_notice( 'tribe-aggregator-facebook-token-expired', array( $this, 'notice_facebook_token_expired' ), 'type=error' );
-		// tribe_notice( 'tribe-aggregator-facebook-oauth-feedback', array( $this, 'notice_facebook_oauth_feedback' ), 'type=success' );
+	/**
+	 * Maybe show notices to users about expiring Facebook Tokens if oauth is enabled
```
Comment (barryhughes) [inline src/Tribe/Aggregator.php]: Kind of related to the comment on line 131: the method name sounds more general purpose to me than something that will/should only ever deal with FB stuff. Should we reword the comment to make it clear Facebook notices are merely its current concern, or else rename the method if the intent is to keep it Facebook-specific?

**Comment by borkweb** [comment]
I've updated the comments as per your suggestions. This is ready for another look.

File: src/Tribe/Aggregator.php [hunk 3]
```diff
@@ -128,13 +128,20 @@ private function __construct() {
...
-		/**
-		 * @todo  fb-reviewed-reactivation
-		 */
-		// tribe_notice( 'tribe-aggregator-facebook-token-expired', array( $this, 'notice_facebook_token_expired' ), 'type=error' );
-		// tribe_notice( 'tribe-aggregator-facebook-oauth-feedback', array( $this, 'notice_facebook_oauth_feedback' ), 'type=success' );
+	/**
+	 * Set up any necessary notices
+	 */
+	public function setup_notices() {
+		if ( ! Tribe__Events__Aggregator::instance()->api( 'origins' )->is_oauth_enabled( 'facebook' ) ) {
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: From: `Tribe__Events__Aggregator::instance()` to `$this`.

File: src/Tribe/Aggregator.php [hunk 4]
```diff
@@ -317,7 +324,8 @@ public static function is_service_active() {
 	 * @return int
 	 */
 	public function get_daily_limit() {
-		return $this->daily_limit;
+		$import_daily_limit = Tribe__Events__Aggregator::instance()->api( 'origins' )->get_limit( 'import' );
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: From: `Tribe__Events__Aggregator::instance()` to `$this`.

File: src/Tribe/Aggregator.php (hunk 3 above)
Comment (borkweb) [inline src/Tribe/Aggregator.php]: Gah. Copy paste fail.

File: src/Tribe/Aggregator.php (hunk 4 above)
Comment (borkweb) [inline src/Tribe/Aggregator.php]: Fixed.

File: src/Tribe/Aggregator.php (hunk 3 above)
Comment (borkweb) [inline src/Tribe/Aggregator.php]: Fixed

**Comment by bordoni** [comment]
:noice: merge away!

**Pull request merged**